test/unittest/rtps/persistence/PersistenceGuidTests.cpp

```diff
@@ -0,0 +1,435 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

Comment:
Looking at the tests, I think they probably fit better into the blackbox category, and they may be split since the use 2 different APIs (DDS and RTPS).

It may be combined with the tests in #1299